File: testapp/java_files/main_hello_name.java
class main_hello_name
{
    public static <E> void check(E expect, E result)
    {
        if(result.equals(expect))
        {
            System.out.println("Correct:\nOutput expected "+expect+" and got "+result);
        }
	else
	{
	    System.out.println("Incorrect:\nOutput expected "+expect+" but got "+result);
	    System.exit(1);
	}
    }
    public static void main(String arg[])
    {
       Test t = new Test();
       String result;
       result = t.hello_name("Raj");
       System.out.println("Input submitted to the function: 'Raj'");
       check("hello Raj", result);
       result = t.hello_name("Pratham");
       System.out.println("Input submitted to the function: 'Pratham'");
       check("hello Pratham", result);
       result = t.hello_name("Ram");
       System.out.println("Input submitted to the function: 'Ram'");
       check("hello Ram", result);
    }
}

File: testapp/exam/c_cpp_files/main_hello_name.c
#include <stdio.h>
#include <stdlib.h>


void check(char expect[], char result[])
{
    if (expect == result)
    {
	printf("Correct:expected %s got %s \n",expect,result);
    }
    else 
    {
	printf("ERROR:expected %s got %s \n",expect,result);
	exit (0);
   }
}

int main(void)
{
	char result[20];
	char A[20]=" pratham";
	char B[20]=" sir";
 	result[20] = message(A);
        printf("%s",result);
	check("hello pratham", result);
	result[20] = message(B);
	check("hello sir",result);
	printf("All Correct\n");
}
